Add pancakeswap to `dex.trades_beta`

tomfutago opened the pull request:
linked to #4759

dune-eng commented:
Workflow run id 7069836729 approved.

dune-eng commented:
Workflow run id 7069895467 approved.

dune-eng commented:
Workflow run id 7069895468 approved.

dune-eng commented:
Workflow run id 7070611278 approved.

dune-eng commented:
Workflow run id 7070699250 approved.

dune-eng commented:
Workflow run id 7070699224 approved.

dune-eng commented:
Workflow run id 7070789364 approved.

dune-eng commented:
Workflow run id 7070789354 approved.

Hosuke commented on models/_sector/dex/trades/ethereum/platforms/pancakeswap_v2_ethereum_base_trades.sql:
Nice hybrid spell.

Hosuke reviewed (approved):
LGTM.✅
Thank you @tomfutago

[pancakeswap base_trades check](https://dune.com/queries/3250162?+dex.trades_beta+_t6c1ea=test_schema.git_dunesql_8a9ab8da_dex_bnb_base_trades&+project+_t6c1ea=%27pancakeswap%27&dex.trades_beta_t6c1ea=test_schema.git_dunesql_c9220ef4_dex_base_trades&project_t6c1ea=%27pancakeswap%27)

jeff-dude commented:
looks like the relationship test is failing? can we add values to the info table and/or fix the gaps?

dune-eng commented:
Workflow run id 7090290401 approved.

dune-eng commented:
Workflow run id 7090290542 approved.

Hosuke reviewed (approved):
Thank you for adding the missing dex.info record.

Hosuke commented on models/_sector/dex/trades/ethereum/_schema.yml:
```diff
@@ -219,4 +219,61 @@ models:
+      - check_seed:
+          seed_file: ref('pancakeswap_ethereum_base_trades_seed')
+          filter:
+            version: 2
+            version: mmpool
```
Oops.

tomfutago replied:
Will have a look at this today.

Hosuke replied:
Solved (partially).

jeff-dude replied:
we need to update this ethereum `version` list similar to the bnb fix

Hosuke commented on macros/test-helpers/check_seed_macro.sql:
```diff
@@ -44,8 +44,12 @@
               {%- if filter is not none %}
-                  {%- for col, val in filter.items() %}
-                      {% if val is not none %} AND seed.{{col}} = '{{val}}' {% endif %}
+                  {%- for col, val_or_vals in filter.items() %}
+                        {% if val_or_vals is iterable and val_or_vals is not string %}
+                            AND seed.{{ col }} IN ({% for val in val_or_vals %}'{{ val }}'{% if not loop.last %}, {% endif %}{% endfor %})
+                        {% else %}
+                            AND seed.{{ col }} = '{{ val_or_vals }}'
+                        {% endif %}
```
I just modified the macro to support multiple `string` values filter.

Hosuke commented on models/_sector/dex/trades/bnb/_schema.yml:
```diff
@@ -112,3 +112,62 @@ models:
+      - check_seed:
+          seed_file: ref('pancakeswap_bnb_base_trades_seed')
+          filter:
+            version:
+              - '2'
+              - 'mmpool'
+              - 'stableswap'
```
This will now compile into `AND seed.version IN ('2', 'mmpool', 'stableswap')`.

tomfutago replied:
Nice! 💪 thanks for fixing this!

Hosuke commented:
I think the `zeroex_polygon_nft_fills_sample` seed's format is illegal and without schema properly defined, I am going to delete this file now and request for a new seed file.

Hosuke commented on seeds/zeroex/polygon/zeroex_polygon_nft_fills_sample.csv:
Hi @RantumBits, since this seed file is missing precision in `*_amount_raw` fields and `block_time` field, I am going to remove this file for now. You can add a new seed file to replace this in another PR if needed.

jeff-dude replied:
changing the seed macro here is running way more than anticipated. one downside to macros for sure.

jeff-dude commented on seeds/_sector/dex/_schema.yml:
```diff
@@ -1185,21 +1185,6 @@ seeds:
         token_sold_amount_raw: uint256
         block_date: timestamp
 
-  - name: pancake_bnb_base_trades_seed
```
was this removed due to a duplicate? or why else are we removing this one only?

dune-eng commented:
Workflow run id 7126061201 approved.

dune-eng commented:
Workflow run id 7126061147 approved.

dune-eng commented:
Workflow run id 7126450226 approved.

dune-eng commented:
Workflow run id 7126450235 approved.

dune-eng commented:
Workflow run id 7126835137 approved.

dune-eng commented:
Workflow run id 7127117766 approved.

dune-eng commented:
Workflow run id 7127117778 approved.

tomfutago commented:
Phew, and I thought this dex will be quick & easy addition..

Few notes on my changes today:
* I removed quotes around each version entry in `_schema` for BNB & ETH as `check_seed_macro` is adding quotes for strings (and all other schema entries are without quotes, so added consistency too). @Hosuke once again thanks for quick fix to this macro!
* there's this model: `nexusmutual_ethereum.capital_pool_eth_daily_transaction_summary` which seems to be broken. I removed seed check on it - to be revisited in a separate PR (maybe original contributor can help @guyhowlett? 🙏)

jeff-dude commented:
> Phew, and I thought this dex will be quick & easy addition..
> 
> Few notes on my changes today:
> 
> * I removed quotes around each version entry in `_schema` for BNB & ETH as `check_seed_macro` is adding quotes for strings (and all other schema entries are without quotes, so added consistency too). @Hosuke once again thanks for quick fix to this macro!
> * there's this model: `nexusmutual_ethereum.capital_pool_eth_daily_transaction_summary` which seems to be broken. I removed seed check on it - to be revisited in a separate PR (maybe original contributor can help @guyhow ? 🙏)

note for future:
if we need to change a macro, especially one that touches on many spells / out of scope of this PR, we should move to new PR and then merge / refresh in original PR. we can see the headaches it can cause

with that said, thanks for continuing to help finalize!

Hosuke commented:
> * I removed quotes around each version entry in `_schema` for BNB & ETH as `check_seed_macro` is adding quotes for strings (and all other schema entries are without quotes, so added consistency too). @Hosuke once again thanks for quick fix to this macro!

Seems the macro will auto wrap values in the filters into string.

jeff-dude commented on seeds/zeroex/polygon/zeroex_polygon_nft_fills_sample.csv:
fyi, i fixed this seed here #4975 
we can merge in main and restore this file

jeff-dude commented:
i'm a bit confused on where we stand on this one. here is my vote:
- open new PR to fix macro for seeds
- move any seeds that are included in this PR solely due to change from seed macro, not anything to do with dex / pancakeswap
- get that merged
- rebase with main here and proceed with clean setup

plz note that #4975 was merged recently to fix one of the 0x seeds, as it was blocking another PR too

jeff-dude commented:
> plz note that #4975 was merged recently to fix one of the 0x seeds, as it was blocking another PR too

while looking into this, i'm piecing together that the `@state:modified` in the `dbt seed` step for CI test is pulling in any seed in the entire dbt lineage of a modified spell, not just those spells modified directly in PR. i believe that is what `@` prefix does. while the use case in this PR is different, we modified the macro directly which causes all to run, it's still good to know for other CI test failure use cases.

[this PR](https://github.com/duneanalytics/spellbook/pull/4887/files) is modifying some static metadata spells, but still kicked off the 0x seed (plus mnay others), since one of the modified spells is in the lineage in which that seeds runs on.

the intention of the `@` is to help get around issues of missing seeds when modifying existing spells that we've hit in CI often (i.e. metadata does not exist failure). working through the kinks!

Hosuke commented:
I will open a new PR for updating the `check_seed` macro first.

Hosuke commented on seeds/dex/offers/dex_offers_seed.csv:
Thank you @tomfutago for fixing so many seed records.
I have moved those changes/fixes into this PR #4978 .
You can have a look to see if I miss anything.

jeff-dude commented:
this one is ready to rebase with main now 👍 

be sure to take the main version of `zeroex_polygon_nft_fills_sample` plz
